src/sentry/event_manager.py

```diff
@@ -1488,41 +1526,64 @@ def _save_aggregate(
                     state=GroupHash.State.LOCKED_IN_MIGRATION
                 ).update(group=group)
 
-                is_new = True
-                is_regression = False
+                is_new = False if seer_matched_group else True
+                is_regression = (
+                    False
+                    if is_new
+                    else _process_existing_aggregate(
+                        # If `seer_matched_group` were `None`, `is_new` would be true and we
+                        # wouldn't be here
+                        group=NonNone(seer_matched_group),
+                        event=event,
+                        incoming_group_values=group_creation_kwargs,
+                        release=release,
+                    )
+                )
```

Comment:
purely opinion based, so entirely up to you, but I find just regular ifs more readable than using conditional expressions multiple times. 


```suggestion
                if is_new := not seer_matched_group:
                    is_regression = False
                    outcome = "new_group"
                else:
                     is_regression =  _process_existing_aggregate(
                        # If `seer_matched_group` were `None`, `is_new` would be true and we
                        # wouldn't be here
                        group=NonNone(seer_matched_group),
                        event=event,
                        incoming_group_values=group_creation_kwargs,
                        release=release,
                    )
                    outcome = "seer_match"
```

and below... 
```python
                span.set_tag("outcome", outcome)
                metric_tags["outcome"] = outcome
```

Reply:
It's true, I do love a good ternary. 🙂  I do sometimes get the sense that they're a little more idiomatic in JS/TS (which is - for better or worse - very much my first love, programming-language-wise) than in Python. Also for whatever reason it actually does make more sense to my brain. But you're right that there's a time and a place, and I do probably lean into it a little hard sometimes.

I'm going to leave it as is for now. It works, and there's lots more to do, but I do take the overall point.

(FWIW, the structure of `_save_aggregate_new` is different enough that when I port this there, I think I am going to have to structure it more with `if`s than with ternaries.)